cpp/src/lists/drop_list_duplicates.cu

```diff
@@ -114,190 +118,160 @@ struct replace_negative_nans_fn {
-                      lists_entries.template begin<Type>(),
-                      lists_entries.template end<Type>(),
+                      input.template begin<Type>(),
+                      input.template end<Type>(),
                       new_entries->mutable_view().template begin<Type>(),
-                      detail::replace_negative_nans_fn<Type>{});
+                      replace_negative_nans_fn<Type>{});
 
     return new_entries;
   }
 
   template <typename Type, std::enable_if_t<std::is_same_v<Type, cudf::struct_view>>* = nullptr>
-  std::unique_ptr<column> operator()(column_view const& lists_entries,
+  std::unique_ptr<column> operator()(column_view const& input,
                                      rmm::cuda_stream_view stream) const noexcept
   {
     std::vector<std::unique_ptr<cudf::column>> output_struct_members;
-    std::transform(
-      thrust::make_counting_iterator(0),
-      thrust::make_counting_iterator(lists_entries.num_children()),
-      std::back_inserter(output_struct_members),
-      [structs_view = structs_column_view{lists_entries}, stream](auto const child_idx) {
-        auto const col = structs_view.get_sliced_child(child_idx);
-        return type_dispatcher(col.type(), detail::replace_negative_nans_dispatch{}, col, stream);
-      });
-
-    return cudf::make_structs_column(lists_entries.size(),
+    std::transform(thrust::make_counting_iterator(0),
+                   thrust::make_counting_iterator(input.num_children()),
+                   std::back_inserter(output_struct_members),
+                   [structs_view = structs_column_view{input}, stream](auto const child_idx) {
+                     auto const col = structs_view.get_sliced_child(child_idx);
+                     return type_dispatcher(
+                       col.type(), replace_negative_nans_dispatch{}, col, stream);
+                   });
+
+    return cudf::make_structs_column(input.size(),
                                      std::move(output_struct_members),
-                                     lists_entries.null_count(),
-                                     cudf::detail::copy_bitmask(lists_entries, stream),
+                                     input.null_count(),
+                                     cudf::detail::copy_bitmask(input, stream),
                                      stream);
   }
 };
 
 /**
- * @brief Generate a 0-based offset column for a lists column.
+ * @brief Generate a 0-based offsets from a lists column.
  *
- * Given a lists_column_view, which may have a non-zero offset, generate a new column containing
- * 0-based list offsets. This is done by subtracting each of the input list offset by the first
- * offset.
+ * Given a lists column which may have a non-zero offset (i.e. the offset values do not start from
+ * `0`), generate a numeric array containing new offset values computed by subtracting the offsets
+ * of the input column by the first offset value.
```

Comment:
It seems unnecessary to actually materialize this instead of just using a transform iterator to create these values on the fly when passing to some other operation.

Reply:
Interesting. Let me try that way.